chain/ethevents/events.go

```diff
@@ -117,6 +126,16 @@ func NewEthEvents(contractsAddresses []common.Address, signer *ethereum.SignKeys
 	if err != nil {
 		return nil, fmt.Errorf("cannot read token storage proof contract abi: %w", err)
 	}
+
+	secureAddrList := make(map[common.Address]bool, len(ethereumWhiteList))
+	for _, sAddr := range ethereumWhiteList {
+		addr := common.HexToAddress(sAddr)
+		if (addr == common.Address{}) || (len(addr) != types.EthereumAddressSize) {
+			return nil, fmt.Errorf("cannot create ethereum whitelist, address %x not valid", addr)
```

Comment:
you should use sAddr in the error message, since addr here could be empty.

and when you do, use %q, since it's already hex-encoded so %x would be double hex encoding.